frontend/src/Registration/index.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+import {
+  View, StyleSheet, Text,
+  ImageBackground, Alert,
+  ActivityIndicator,
+} from 'react-native';
+import { useSelector, useDispatch } from 'react-redux';
+import { useNavigation } from 'react-navigation-hooks';
+import { registrationSuccess } from './actionCreator';
+
+import validation from '../common/helper';
+import Colors from '../common/colors';
+import background from '../../assets/login/background.jpg';
+import { InputField, SubmitButton } from '../common/components';
+
+const styles = StyleSheet.create({
+  container: {
+    alignItems: 'center',
+    justifyContent: 'center',
+    padding: 0,
+  },
+  header: {
+    width: 300,
+    marginVertical: 12,
+    fontSize: 24,
+    fontWeight: 'bold',
+    color: Colors.whiteColor,
+  },
+  background: {
+    width: '100%',
+    height: '100%',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  button: {
+    flexDirection: 'row',
+    marginTop: 20,
+    justifyContent: 'space-between',
+    width: '83%',
+  },
+});
+
+function showAlert(text) {
+  Alert.alert('Warning', text);
+}
+
+function Registration() {
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+  const isLoading = useSelector((state) => state.registration.isLoading);
+  const registerError = useSelector((state) => state.registration.error);
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [kingdomName, setKingdomName] = useState('');
+  const USERNAME_BLACKLIST = ['admin', 'root'];
+
+  function handleSubmit() {
+    const name = username.toLowerCase();
+    if (name === '' || email === '' || password === '') {
+      return showAlert('All the input fields are required.');
+    }
+    if (password.length <= 7) {
+      return showAlert('Password must be at least 8 characters');
+    }
+    if (!validation(email)) {
+      return showAlert('Please reenter a valid email');
+    }
+    if (USERNAME_BLACKLIST.includes(name.toLowerCase())) {
+      return showAlert('Please reenter a valid username');
+    }
+    if (!kingdomName) {
+      setKingdomName(`${name}'s Kingdom`);
+    }
+    dispatch(registrationSuccess());
+    if (registerError !== undefined) {
+      return showAlert(`Oops, ${registerError.message}`);
+    }
+    if (isLoading) {
+      return <ActivityIndicator size="large" color={Colors.tealColor} />;
+    }
+    showAlert('Registration Success.');
+    return navigation.navigate('Map');
```

Comment:
this part doesn't work for me, neither on Android or on iPhone, I just get a warning about the successful registration, but it doesn't redirect me to the mocked Map page, is it working on a simulator or tablet? can you please check it?